src/meet-calendar.ts

```diff
@@ -0,0 +1,688 @@
+    return this.element?.text() ?? "";
+  }
+
+  /**
+   * Adds text to the description.
+   * @param text
+   */
+  addDescriptionText(text: string) {
+    let el = this.element;
+    if (!el) return;
+
+    let descriptionNode = el[0];
+    descriptionNode.dispatchEvent(getKeyboardEvent("keydown"));
+    descriptionNode.dispatchEvent(getKeyboardEvent("keypress"));
+
+    // format new lines
+    let textToInsert = text.replace(/(?:\r\n|\r|\n)/g, "<br />");
+
+    // if there is already text in the description append on new line
+    if (el.text().length > 0) {
+      el.append("<br/><br/>");
+    }
+    el.append(textToInsert);
+
+    descriptionNode.dispatchEvent(getKeyboardEvent("input"));
+    descriptionNode.dispatchEvent(getKeyboardEvent("keyup"));
+  }
+
+  /**
+   * Updates the initial button text and click handler when there is
+   * no meeting scheduled.
+   */
+  updateInitialButtonURL(location: LocationWrapper) {
+    let button = $("#jitsi_button");
+    button.html("Add a " + LOCATION_TEXT);
+
+    let container = this.event.buttonContainer;
+    if (container) {
+      container.parent().off("click");
+      container.parent().on("click", (e) => {
+        e.preventDefault();
+
+        this.clickAddMeeting(false, location);
+      });
+    }
+  }
+
+  /**
+   * Updates the url for the button.
+   */
+  updateButtonURL() {
+    try {
+      var button = $("#jitsi_button");
+      button.html("Join your " + LOCATION_TEXT + " now");
+
+      var container = this.event.buttonContainer;
+
+      if (container) {
+        container.parent().off("click");
+        container.parent().on("click", (e) => {
+          e.preventDefault();
+
+          // call updateMeetingId, the case where somebody edited location
+          // and then click join now before saving
+          this.event.updateMeetingId();
+
+          window.open(BASE_URL + this.event.meetingId, "_blank");
+        });
+      }
+    } catch (e) {
+      console.log(e);
+    }
+  }
+}
+
+/**
+ * Returns an event object that can be used to be simulated
+ */
+function getKeyboardEvent(eventType: string) {
+  return new KeyboardEvent(eventType, {
+    bubbles: true,
+    cancelable: true,
+    view: window,
+    ctrlKey: false,
+    altKey: false,
+    shiftKey: false,
+    metaKey: false,
+    keyCode: 32,
+    charCode: 0,
+  });
+}
+
+/**
+ * Finds a parameter in the page url parameters.
+ * @param parameterName the name of the param to search for
+ * @returns {String} the parameter value.
+ */
+function findGetParameter(parameterName: string): string | null {
+  var result = null,
+    tmp = [];
+  location.search
+    .substr(1)
+    .split("&")
+    .forEach(function (item) {
+      tmp = item.split("=");
+      if (tmp[0] === parameterName) result = decodeURIComponent(tmp[1]);
+    });
+  return result;
+}
+
+/**
+ * Checks whether it is ok to add the button to current page
+ * in case of new google calendar interface
+ */
+function checkAndUpdateCalendarG2() {
```

Comment:
seems like this needs a test, preferably one that loads gcal in puppeteer or some other browser for test purposes.

Reply:
tests added in later updates, admittedly using stub html rather as that enables them to be run reliably during the CI build